File: rb/features.rb
# YuGiOh SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module YuGiOhFeatures
  def self.make_feature(name)
    case name
    when "base"
      YuGiOhBaseFeature.new
    when "ratelimit"
      YuGiOhRatelimitFeature.new
    when "retry"
      YuGiOhRetryFeature.new
    when "test"
      YuGiOhTestFeature.new
    when "timeout"
      YuGiOhTimeoutFeature.new
    else
      YuGiOhBaseFeature.new
    end
  end
end

File: py/features.py
# YuGiOh SDK feature factory

from feature.base_feature import YuGiOhBaseFeature
from feature.test_feature import YuGiOhTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YuGiOhBaseFeature(),
        "test": lambda: YuGiOhTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
